csrc/multidevice/executor.cpp

```diff
@@ -226,7 +190,12 @@ std::vector<at::Tensor> MultiDeviceExecutor::runWithInput(
       inputs.size() == staged_fusion_->inputs().size(),
       "Wrong number of inputs");
 
-  val_to_IValue_ = allocateRecvBuffers(inputs);
+  auto allocations =
```

Comment:
IIUC, we are allocating all the recv buffers separately even when, e.g., one buffer could be reused, right? 

Have you considered reusing buffers?

Reply:
That's correct! No, there is no support for reusing buffers.
We also do not support inplace collective for now, so there is allocation also when the receiver is the sender as well.

These optimizations could be done later-on, and then the patch will be quit orthogonal to this pr, only affecting the content of `vals_to_allocate_`

Reply:
I added a comment and a TODO in the `MultiDeviceExecutor`'s constructor